src/Symfony/Bridge/Doctrine/Security/User/EntityUserProvider.php

```diff
@@ -51,7 +51,7 @@ public function loadUserByUsername($username)
             $user = $repository->findOneBy(array($this->property => $username));
         } else {
             if (!$repository instanceof UserProviderInterface) {
-                throw new \InvalidArgumentException(sprintf('The Doctrine repository "%s" must implement UserProviderInterface.', get_class($repository)));
+                throw new \InvalidArgumentException(sprintf('You must either make the Doctrine Repository of the entity "%s" ("%s") implement "Symfony\Component\Security\Core\User\UserProviderInterface" or set the "property" option in the corresponding entity provider configuration.', $this->classOrAlias, get_class($repository)));
```

Comment:
👍  I hope I correctly understood you (`... the "[entity]" entity Doctrine Repository ("[repo]")`).